### DISCUSSION TITLE
How to pass string to connection parameters on initializing

### DISCUSSION THREAD

**Discussion by ammarabbasi1986** [tdslite/tdslite#38] [Q&A] [answered]
Please guide, how to pass string to connection parameters as bellow.
Platformio.

e.g.
**String sql_user = "user";**

    params.server_name = TDSL_PMEMSTR("192.168.1.22"); // WL
    // SQL server port number
    params.port        = 14333; // default port is 1433
    // SQL server login user
    **params.user_name   = user;**
    // SQL server login user password
    params.password    = TDSL_PMEMSTR("2022-tds-lite-test!");
    // Client name(optional)
    params.client_name = TDSL_PMEMSTR("arduino uno");
    // App name(optional)
    params.app_name    = TDSL_PMEMSTR("initialize library example");
    // Database name(optional)
    params.db_name     = TDSL_PMEMSTR("master");

Thanks.

**Comment by ammarabbasi1986** [comment]
Can i use this method?

String sql_user;
tdsl::string_view str_user {sql_user.c_str(), sql_user.size()};
params.user_name   = str_user;

**Reply by xmkg** [reply, marked as answer]
Hi @ammarabbasi1986 yes, you should be able to use that.

**Reply by ammarabbasi1986** [reply]
Thanks @mustafakemalgilor , how can we pass const char * to params.user_name ? please guide.

i will check this and update, away for few days.

**Reply by ammarabbasi1986** [reply]
Thanks, it worked, have to change driver declaration through:

//decltype(driver)::progmem_connection_parameters params;
   decltype(driver)::connection_parameters params;
